### PR TITLE
Fetch data from server API

### PR DESCRIPTION
**Pull request by jnbooth**
### Setup

1. Copy [`.env.example`](https://github.com/Team-Asteraceae/project-catwalk/blob/api/.env.example) to a new file named `.env`
2. In `.env`, fill in the `VITE_API_TOKEN` field with a [GitHub token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
3. To start using data from the API, access your component's props (see comment below)

### Modified React components
- App
  3 new properties:
  - `info: ProductInfo` - information about a product
  - `related: ProductInfo[]` - information about related items
  - `reviews: Review[]` - list of all reviews for a product
- RatingsAndReviews
  2 new properties:
  - `reviews: Review[]` - list of all reviews for a product
  - `metadata: Metadata` - review metadata for a product
- RelatedItemsAndComparisons
  1 new property:
  - `products: ProductInfo[]` - information about related items

### Functionality
Upon initialization, the app detects the current product from the URL. For example, `127.0.0.1/40350` displays product information for product #40350. For testing purposes only, if no product is specified in the URL, the app defaults to product #40344, "Camo Onesie".

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+VITE_API=https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/
+VITE_API_TOKEN=
```
Comment (jnbooth) [inline .env.example:2]: Make sure to copy this file to `.env` and fill this field in with your own GitHub token!

File: src/ratings-and-reviews/components/ReviewList.jsx [hunk 2]
```diff
@@ -3,19 +3,19 @@ import ReviewTile from './ReviewTile.jsx';
 
 
 
-const ReviewList = ({reviewList}) => {
+const ReviewList = ({reviews}) => {
```
Comment (jnbooth) [inline src/ratings-and-reviews/components/ReviewList.jsx:6]: This prop has been renamed from `reviewList` to `reviews` for consistency

File: src/ratings-and-reviews/index.jsx [hunk 3]
```diff
@@ -19,35 +18,21 @@ const reviewsTotal = (reviews) => {
...
-    count += reviews[i];
-  }
-  return parseFloat(total / count).toFixed(1);
-};
-
-let averageRating = reviewsAverage(reviewMetaData.ratings);
-
-
-const RatingsAndReviews = () => (
+const RatingsAndReviews = (/* { reviews, metadata } */) => (
```
Comment (jnbooth) [inline src/ratings-and-reviews/index.jsx:21]: @ajpsyk If at some point you want to start using API data, simply uncomment these props so you get:
```js
const RatingsAndReviews = ({ reviews, metadata }) => (
```
At that point you can also remove the `./sampleData.js` import.

File: src/ratings-and-reviews/index.jsx [hunk 4]
```diff
@@ -19,35 +18,21 @@ const reviewsTotal = (reviews) => {
   return total;
 };
 
-/** @param {Object} reviews*/
-const reviewsAverage = (reviews) => {
```
Comment (jnbooth) [inline src/ratings-and-reviews/index.jsx:23]: This is calculated up front by the API, so no need to do it again here

File: src/related-items-and-comparisons/RelatedItemsList.jsx [hunk 5]
```diff
@@ -31,12 +31,14 @@ const RelatedItemsList = (props) => {
       <div className="related-container">
         {leftButton}
         <div className="related-items-list">
-          {currList.map((product) => <CardItem key={product.id} product={product} />)}
+          {currList.map(({ product, metadata, styles }) =>
```
Comment (jnbooth) [inline src/related-items-and-comparisons/RelatedItemsList.jsx:34]: @jonathanjewett Your info for each item now includes metadata and styles.

File: src/related-items-and-comparisons/index.jsx [hunk 6]
```diff
@@ -3,9 +3,9 @@ import CardItem from './CardItem.jsx';
 import RelatedItemsList from './RelatedItemsList.jsx';
 import './related-items-and-comparisons.scss';
 
-import { products, stylesForProducts, relatedProducts } from './sampleData.js';
+import { products } from './sampleData.js';
 
-const RelatedItemsAndComparisons = (props) => (
+const RelatedItemsAndComparisons = (/* { products } */) => (
```
Comment (jnbooth) [inline src/related-items-and-comparisons/index.jsx:8]: @jonathanjewett If at some point you want to start using API data, simply uncomment these props so you get:

```js
const RelatedItemsAndComparisons = ({ products }) => (
```

At that point you can also remove the ./sampleData.js import.

File: typings/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    characteristics: {[characteristic: string]: {id: number, value: number}},
+    rating: number
+}
+
+// Union of database lookups
+interface ProductInfo {
+    product: Product,
+    metadata: Metadata,
+    styles: Style[],
+}
```
Comment (jnbooth) [inline typings/index.d.ts:84]: This file is your quick reference for the shape of the data.

**Comment by jnbooth** [comment]
[I want to retrieve info from the API server](https://trello.com/c/nR1Yiego/85-i-want-to-retrieve-info-from-the-api-server)

File: src/related-items-and-comparisons/RelatedItemsList.jsx (hunk 5 above)
Comment (jonathanjewett) [reply src/related-items-and-comparisons/RelatedItemsList.jsx:34]: Thanks, yeah it's probably better to have too much than too little, I can always filter what I need.

File: src/related-items-and-comparisons/index.jsx (hunk 6 above)
Comment (jonathanjewett) [reply src/related-items-and-comparisons/index.jsx:8]: Acknowledged

**Pull request merged**